includes/export/class-wc-product-csv-exporter.php

```diff
@@ -176,9 +176,11 @@ public function prepare_data_to_export() {
 		$this->row_data    = array();
 		$variable_products = array();
 
+		$fetch_variations = apply_filters( "woocommerce_product_export_{$this->export_type}_fetch_variations", ! empty( $args['category'] ), $args );
```

Comment:
Please change it to:

```php
	/**
		* Filter product is variable.
		*
		* It checks if is a variable product and if the category is set,
		* this means we need to fetch variations separately as they are not tied to a category.
		*
		* @param bool       $is_variable If product is variable.
		* @param WC_Product $product     Product instance.
		* @param array      $args        Export args.
		*/
	$is_variable = apply_filters( "woocommerce_product_export_{$this->export_type}_is_variable", $product->is_type( 'variable' ) && ! empty( $args['category'] ), $product, $args );
```

Reply:
Hey @claudiosanches thanks for the review.

I was going to apply your suggestion but in practice I think you didn't realize that this way we would have to move the filter to inside the loop, as you relied on the single product at `$product->is_type( 'variable' ) && ! empty( $args['category'] )`

What you think? Is this an issue?

![Screenshot from 2020-04-04 11-34-00](https://user-images.githubusercontent.com/10077439/78453481-5ac8b200-7668-11ea-8d81-695b9a67af8f.png)
Here is a screenshot with your suggestions implemented

Reply:
Also about the filter's name, we are actually checking if the product variations should be also requested in a subsequent query. The name you suggested `_is_variable` is not always the real meaning.

We could have a case where the product is variable but we don't want the subsequent query because the initial query already includes them.

In this case the semantics is weird: if the product is variable but the category is not given as a filter, the variable `$is_variable` would be `false`, and if we implement a custom filter that requires variations to be fetch after, we would need to hook into `_is_variable` to force it to `true`

If you want to make it clear that we are working with variable products only, I suggest something like: 
`"woocommerce_product_export_{$this->export_type}_query_product_variations_separately"`

Reply:
Yes, probably better to include in the loop, so you can check individually for each product.

> If you want to make it clear that we are working with variable products only, I suggest something like:
"woocommerce_product_export_{$this->export_type}_query_product_variations_separately"

We are not querying products, we are checking if is a variable product.